jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -266,13 +266,15 @@
     boolean isOptimizedForDirectBuffers();
 
     /**
-     * Upgrade connections.
-     * Close the old connection, update the endpoint and open the new connection.
-     * If the oldConnection is an instance of {@link Connection.UpgradeFrom} then
-     * a prefilled buffer is requested and passed to the newConnection if it is an instance
-     * of {@link Connection.UpgradeTo}
+     * <p>Upgrades this EndPoint from the current connection to the given new connection.</p>
+     * <p>Closes the current connection, links this EndPoint to the new connection and
+     * then opens the new connection.</p>
+     * <p>If the current connection is an instance of {@link Connection.UpgradeFrom} then
+     * a buffer of unconsumed bytes is requested; if there are unconsumed bytes, and if
+     * the new connection is an instance of {@link Connection.UpgradeTo}, the unconsumed
+     * buffer is passed to the new connection.</p>
```

Comment:
Shouldn't we also specify what happens when the current connection is not an instance of `UpgradeFrom` and/or the new connection is not an instance of `UpgradeTo`?

Also, we could specify what happens when `onUpgradeTo` does not consume all the bytes of the buffer, like throwing an exception, or discarding the bytes with a warning.